sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java

```diff
@@ -27,4 +27,11 @@
         // This is intentionally left blank. This constructor exists
         // only to prevent external assemblies inheriting from it.
     }
+/*
```

Comment:
Thoughts on adding this section to RuleFilter?
Ideally, all rule filters should implement their specific equals and hashcode methods, so it would make sense to add this here and force new RuleFilters to add them.

Reply:
Hmm, I didn't think you would be able to make those methods abstract, as they are already implemented in Object.java (the base object).  I guess if that works, sure (RuleFilter would need to be defined as an abstract class, however).

My suspicion is the compiler will complain about making those methods abstract.  If that is the case, it would be better off continuing to default to the inherited implementations and just ensure that the subclasses of RuleFilter override with appropriate implementations.

Reply:
Hmm, I am not sure why it won't be possible, but I just added them in and it compiled without any errors

Reply:
Then I learned something new today. :)